server/src/test/java/org/opensearch/snapshots/SnapshotResiliencyTests.java

```diff
@@ -686,6 +687,107 @@ public void testConcurrentSnapshotCreateAndDeleteOther() {
         }
     }
 
+    public void testTransportGetSnapshotsAction() {
+        setupTestCluster(randomFrom(1, 3, 5), randomIntBetween(2, 10));
+
+        String repoName = "repo";
+        final String[] snapshotsList = { "snapshot-1", "snapshot-2" };
+        final String[] indexList = { "index-1", "index-2" };
+        final int shards = randomIntBetween(1, 10);
+
+        TestClusterNodes.TestClusterNode clusterManagerNode = testClusterNodes.currentClusterManager(
+            testClusterNodes.nodes.values().iterator().next().clusterService.state()
+        );
+
+        for (int i = 0; i < snapshotsList.length; i++) {
+            final StepListener<CreateSnapshotResponse> createSnapshotResponseStepListener = new StepListener<>();
+            final String snapshot = snapshotsList[i];
+            final String index = indexList[i];
+            continueOrDie(
+                createRepoAndIndex(repoName, index, shards),
+                createIndexResponse -> client().admin()
+                    .cluster()
+                    .prepareCreateSnapshot(repoName, snapshot)
+                    .setWaitForCompletion(true)
+                    .execute(createSnapshotResponseStepListener)
+            );
+
+            final StepListener<GetSnapshotsResponse> getSnapshotsResponseStepListener = new StepListener<>();
+            continueOrDie(getSnapshotsResponseStepListener, getSnapshotsResponse -> {
+                client().admin().cluster().prepareGetSnapshots(repoName).execute(getSnapshotsResponseStepListener);
+            });
```

Comment:
why are we doing this?

Reply:
This is required for the listener based calls. as createSnapshot is following as well, else the repositoryData will be empty.

Reply:
i dont see any link of below getRepositoryData with this extra get call.

Reply:
This is not required anymore.